paper-api/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java

```diff
@@ -143,49 +155,118 @@ public Player getEntity() {
 
     // Paper end - improve death events
 
-    // Paper start - adventure
+    /**
+     * Get whether the death message should be shown.
+     * By default, this is determined by {@link org.bukkit.GameRule#SHOW_DEATH_MESSAGES}.
+     *
+     * @return whether the death message should be shown
+     * @see #deathChatMessage()
+     * @see #deathScreenMessage()
+     */
+    public boolean shouldDisplayDeathMessage() {
+        return displayDeathMessage;
+    }
+
+    /**
+     * Set whether the death message should be shown.
+     * By default, this is determined by {@link org.bukkit.GameRule#SHOW_DEATH_MESSAGES}.
+     *
+     * @param displayDeathMessage whether the death message should be shown
+     * @see #deathChatMessage()
+     * @see #deathScreenMessage()
+     */
+    public void setShouldDisplayDeathMessage(boolean displayDeathMessage) {
+        this.displayDeathMessage = displayDeathMessage;
+    }
+
     /**
      * Set the death message that will appear to everyone on the server.
      *
      * @param deathMessage Component message to appear to other players on the server.
+     * @deprecated in favour of {@link #deathChatMessage(Component)} and {@link #deathScreenMessage(Component)}
      */
-    public void deathMessage(final net.kyori.adventure.text.@Nullable Component deathMessage) {
-        this.deathMessage = deathMessage;
+    @Deprecated
```

Comment:
Why does this need to be deprecated? Seems like a good shortcut if you want to set both messages to the same value instead of calling the two new methods.